### ISSUE TITLE
[Next.js Example] CSSReset is not inside ColorModeProvider

### ISSUE THREAD

**Issue by kolyasapphire** [chakra-ui/chakra-ui#382]
**Describe the bug**
CSSReset doesn't style `<html>` according to theme changes (light/dark) if it's outside `<ColorModeProvider>`

**Suggested solution(s)**
Modify the example from:
```
      <ThemeProvider theme={theme}>
        <CSSReset />
        <ColorModeProvider>
          <Component />
        </ColorModeProvider>
      </ThemeProvider>
```
To:
```
      <ThemeProvider theme={theme}>
        <ColorModeProvider>
          <CSSReset />
          <Component />
        </ColorModeProvider>
      </ThemeProvider>
```
I'd be happy to submit a pull request if needed.

**Comment by kolyasapphire** [comment]
Hi @segunadebayo is this not a valid issue?

**Comment by jrrmt** [comment]
I had this issue as well, since CSSReset must set e.g. base text colors, it should be inside the ColorModeProvider.

For example, Input text color would be wrong otherwise.

**Comment by with-heart** [comment]
I believe this was closed because the `next` example has been fixed in `dev-ts` and will be correct for 1.0. It's not mentioned in the 1.0 docs yet though, so I'm going to reopen this so we don't forget to mention it there as well.

**Comment by with-heart** [comment]
This is now handled correctly using `ChakraProvider` in 1.0 (currently available as an alpha release)!

```jsx
import {ChakraProvider} from '@chakra-ui/core'
import {theme} from '@chakra-ui/theme'

function App() {
  return (
    <ChakraProvider theme={theme}>
      <CSSReset />
      <div>Hello World!</div>
    </ChakraProvider>
  )
}
```

Usage as it relates to this issue is documented in the first sentence of https://next.chakra-ui.com/docs/color-mode.
